oak/server/oak_node.cc

```diff
@@ -143,4 +147,29 @@ bool OakNode::WaitOnChannels(std::vector<std::unique_ptr<ChannelStatus>>* status
   }
 }
 
+void OakNode::AddLabelsToMessage(Message* message) const {
```

Comment:
Alternatively, we could have a method on `Labels` that merges two instances, which would probably mean union of secrecy tags and intersection of integrity tags? Then here you would extract labels from the message, and merge them with the node labels, and assign back to the node. Or maybe we need two operations (corresponding to `meet` and `join` in a hypothetical label lattice)?